src/app/(app)/purchases.tsx

```diff
@@ -0,0 +1,69 @@
+import { useFocusEffect } from 'expo-router';
+import { useCallback, useEffect, useState } from 'react';
+import { FlatList } from 'react-native';
+
+import type { CartItem } from '@/api/cart/types';
+import { API_CONSTS } from '@/api/consts';
+import { usePruchases } from '@/api/purchase/get-purchases';
+import type { Purchase } from '@/api/purchase/types';
+import { HeaderLogo } from '@/components/header-logo';
+import { PurchaseCard } from '@/components/purchases/purchase-card';
+import { SafeAreaView, Text, View } from '@/ui';
+
+export default function Purchases() {
+  const { data, hasNextPage, fetchNextPage, isFetchingNextPage, refetch } =
+    usePruchases({
+      variables: { items: API_CONSTS.INITIAL_ITEMS },
+    });
+
+  const [lineItems, setLineItems] = useState<CartItem[]>([]);
+  const handleReachEnd = () => {
+    if (hasNextPage && !isFetchingNextPage) {
+      fetchNextPage();
+    }
+  };
+
+  useFocusEffect(
+    useCallback(() => {
+      refetch();
+    }, [refetch])
+  );
+
+  useEffect(() => {
+    const allLineItems =
+      data?.pages.flatMap((page) =>
+        page.data.flatMap((purchase: Purchase) => purchase.line_items)
+      ) || [];
+    setLineItems(allLineItems);
+  }, [data]);
```

Comment:
This effect can be removed, [as reference](https://react.dev/learn/you-might-not-need-an-effect).

I identified two alternatives to get rid of it:
* [select](https://tanstack.com/query/latest/docs/framework/react/reference/useQuery#usequery) prop in the get-purchases infinite query. There, you can transform your data without worrying if it exists; the react query will validate it for you.
* Transforming the data during render, without saving it into state.